docs/topics/data-analysis/data-analysis-notebooks-output-formats.md

```diff
@@ -1,35 +1,293 @@
 
-There will be examples for each output.
+Kotlin Notebook supports a variety of output types, including text, HTML, and images. With the help of external libraries,
+you can expand your output options and visualize your data with charts, spreadsheets, and more.
 
-One key feature that makes notebooks stand out is the variety of output types they support:
+Each output is a JSON object that maps the [Jupiter MIME type](https://jupyterlab.readthedocs.io/en/latest/user/file_formats.html)
+to some data. From this map, Kotlin Notebook picks the MIME type that it supports and has the highest priority among other
+types and renders it like this:
 
-## Text
+* [Text](#texts) uses `text/plain` MIME type.
+* The [BufferedImage class](#buffered-images) serializes images into a Base64 string and uses the `image/png` mime type.
+* The [Image class](#loaded-images), as well as the [LaTeX format](#math-formulas-and-equations), use the `text/html` MIME
+  type with the `img` tag inside.
+* [Kotlin DataFrame tables](#data-frames) and [Kandy plots](#charts) use their own internal MIME types, which are backed
+  by static `HTML/images`. This way, you can display them on GitHub.
 
-This is the simplest type of output, and it includes printed statements, variable values, or any text-based output from your code. If a cell’s result doesn’t fall under one of the categories below, it will be printed as text via the toString() method.
+You can set up the mapping manually, for example, to use Markdown as a cell output:
 
-## Rich text
+```kotlin
+MimeTypedResult(
+    mapOf(
+        "text/plain" to "123",
+        "text/markdown" to "# HEADER",
+        //other mime:value pairs
+    )
+)
+```
 
-Markdown cells produce rich text HTML output, offering support for lists, font styles, code blocks, and more.
+To display any kind of output, use the `DISPLAY()` function. It also enables the combination of several outputs:
 
-## HTML markup
+```kotlin
+DISPLAY(HTML("<h2>Gaussian distribution</h2>"))
+DISPLAY(LATEX("f(x) = \\frac{1}{\\sigma \\sqrt{2\\pi}} \\cdot e^{-\\frac{(x - \\mu)^2}{2\\sigma^2}}"))
 
-Jupyter notebooks can render HTML directly, enabling rich text formatting, the use of tables, or even the embedding of websites.
+val experimentX = experimentData.map { it.key }
+val experimentY = experimentData.map { it.value }
+
+DISPLAY(plot {
+    bars {
+        x(experimentX)
+        y(experimentY)
+    }
+})
+```
+
+![Different outputs for Gaussian distribution](gaussian-distribution-output.png){width=700}
+
+## Texts
+
+### Plain text
+
+The simplest output type is plain text. It's used in printed statements, variable values, or any text-based output from
+your code:
+
+```kotlin
+val a1: Int = 1
+val a2: Int = 2
+var a3: Int? = a1 + a2
+
+"My answer is $a3"
+```
+
+![Plain text code output](plain-text-output.png){width=300}
+
+* If a cell's result doesn't match with other output types, it will be printed as plain text using the `toString()` function.
```

Comment:
It may be not crystal clear what "doesn't match" mean, maybe worths rephrasing it a bit. Here's a description of what's happening under the hood, you can add this link if you wish
https://github.com/Kotlin/kotlin-jupyter?tab=readme-ov-file#rendering